site/scripts/lintChangesets.ts

```diff
@@ -0,0 +1,66 @@
+/* eslint-disable no-console */
+import { glob, readFile } from 'fs/promises';
+import { resolve } from 'path';
+import { styleText } from 'util';
+
+import { supportedLanguages } from '../src/App/Code/supportedLanguages.ts';
+
+const braidChangesetRegex =
+  /^---\n'braid-design-system': (?:patch|minor|major)\n---$/m;
+
+const markdownCodeBlockLanguageRegex = /``` *(?<language>[a-z]*)\n/g;
+
+const info = (s: string) => styleText('blue', s);
+const critical = (s: string) => styleText('red', s);
+
+const repoRoot = resolve(import.meta.dirname, '../../');
+
+const errorMessages: string[] = [];
+
+for await (const entry of glob('.changeset/!(README).md', {
+  cwd: repoRoot,
+})) {
+  console.log(`Checking changeset: ${info(entry)}`);
+
+  const content = await readFile(resolve(repoRoot, entry), 'utf-8');
+
+  const isBraidChangeset = braidChangesetRegex.test(content);
+  if (!isBraidChangeset) {
+    console.log(`Skipping changeset: ${info(entry)} (not a Braid changeset)`);
+    continue;
+  }
+
+  let backtickCounter = 0;
+  const matches = content.matchAll(markdownCodeBlockLanguageRegex);
+
+  for (const match of matches) {
+    // Naively skip closing backticks
+    backtickCounter++;
+    if (backtickCounter % 2 === 0) {
+      continue;
+    }
+
+    // Skip code blocks that have no language
+    const language = match.groups?.language;
+    if (!language) {
+      continue;
+    }
```

Comment:
If we update the regex above (and allow no language as we do here), these checks are no longer required, as we just check the matching languages exist in the `supportedLanguages`. Thoughts?

Reply:
Yeah good shout. I was originally capturing all backticks regardless, but since we don't care about blocks with no language this makes sense.